releng/run.sh

```diff
@@ -14,16 +14,22 @@ else
   exit 1
 fi
 
-# Ensure that virtualenv is installed and updated
-$PYTHON_CMD -m pip install --quiet --user --upgrade --requirement "$DIR/requirements_virtualenv.txt"
-
-# Create and activate virtualenv
-VENV="$DIR/.virtualenv"
-$PYTHON_CMD -m virtualenv --quiet "$VENV" > /dev/null
-# Since virtualenv is made by idiots, temporarily disable unbound variable checks when activating virtualenv
-set +o nounset
+VENV="$DIR/.venv"
+
+# Create virtual environment
+if [[ ! -d "$VENV" ]]
+then
+    if ! $PYTHON_CMD -m venv "$VENV" > /dev/null
+    then
+        # On Ubuntu/Debian, Python3 does not automatically come with ensurepip installed, so we install it manually:
+        # https://bugs.launchpad.net/ubuntu/+source/python3.4/+bug/1290847/+index?comments=all (comment #58)
+        $PYTHON_CMD -m venv --without-pip "$VENV" > /dev/null
+        curl -s https://bootstrap.pypa.io/get-pip.py | "$VENV/bin/python" > /dev/null
```

Comment:
Downloading a script from the internet and running it is out of the question as part of the `./b` script, because this would be done without the user's knowledge/consent. Could you detect if pip is installed or not, and ask the user to install it?

Reply:
Fixed! Now, if the `venv` cannot be created directly because of a missing `ensurepip` module, the user should fix this themselves before they can run the `./b` script. I printed some options in the console so that the user can choose between globally installing some extra stuff (installing `python3-venv`) or trust the `get-pip.py` script to install Pip into the virtual environment.

If `venv` fails for any other reason, `venv` will print its own error message.